x-pack/plugins/ml/public/application/components/ml_inference/add_inference_pipeline_flyout.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useEffect, useMemo, useState } from 'react';
+
+import {
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { ModelItem } from '../../model_management/models_list';
+import { AddInferencePipelineSteps } from './types';
+import { AddInferencePipelineFooter } from './components/add_inference_pipeline_footer';
+import { AddInferencePipelineHorizontalSteps } from './components/add_inference_pipeline_horizontal_steps';
+import { getInitialState, getModelType } from './state';
+import { ConfigurePipeline } from './components/configure_pipeline';
+import { AdvancedConfiguration } from './components/advanced_configuration';
+import { TestPipeline } from './components/test_pipeline';
+import { ReviewPipeline } from './components/review_and_create_pipeline';
+import { useMlApiContext, useMlKibana } from '../../contexts/kibana';
+import { getPipelineConfig } from './get_pipeline_config';
+import {
+  validateInferencePipelineConfigurationStep,
+  validateInferencePipelineAdvancedStep,
+} from './validation';
+import type { MlInferenceState, InferenceModelTypes } from './types';
+
+import './add_inference_pipeline_flyout.scss';
+
+export interface AddInferencePipelineFlyoutProps {
+  onClose: () => void;
+  model: ModelItem;
+}
+
+export const AddInferencePipelineFlyout: FC<AddInferencePipelineFlyoutProps> = ({
+  onClose,
+  model,
+}) => {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const initialState = useMemo(() => getInitialState(model), [model.model_id]);
+  const [formState, setFormState] = useState<MlInferenceState>(initialState);
+  const [step, setStep] = useState<AddInferencePipelineSteps>(
+    AddInferencePipelineSteps.Configuration
+  );
+  const [pipelineNames, setPipelineNames] = useState<string[]>([]);
+
+  const {
+    trainedModels: { getAllIngestPipelines, createInferencePipeline },
+  } = useMlApiContext();
+  const {
+    notifications: { toasts },
+  } = useMlKibana();
+  const modelType = getModelType(model);
+
+  const createPipeline = async () => {
+    setFormState({ ...formState, creatingPipeline: true });
+    try {
+      await createInferencePipeline(formState.pipelineName, getPipelineConfig(formState));
+      setFormState({ ...formState, pipelineCreated: true, creatingPipeline: false });
+    } catch (e) {
+      setFormState({ ...formState, creatingPipeline: false, pipelineError: e.message });
+    }
+  };
+
+  useEffect(() => {
```

Comment:
This could be moved into a custom hoook `useFetchPipelines()` and return the pipelines instead of managing the state inline in this component.

Reply:
Updated in 4b1791c5523f98483b24f0ca0dd3f1e88b855a50